crypto/src/hash/rescue/rp64_256/mod.rs

```diff
@@ -218,102 +218,116 @@ impl ElementHasher for Rp64_256 {
-    let mut result = [BaseElement::ZERO; STATE_WIDTH];
-    result.iter_mut().zip(MDS).for_each(|(r, mds_row)| {
-        state.iter().zip(mds_row).for_each(|(&s, m)| {
-            *r += m * s;
+    #[inline(always)]
+    fn apply_mds(state: &mut [BaseElement; STATE_WIDTH]) {
+        let mut result = [BaseElement::ZERO; STATE_WIDTH];
+        result.iter_mut().zip(MDS).for_each(|(r, mds_row)| {
+            state.iter().zip(mds_row).for_each(|(&s, m)| {
+                *r += m * s;
+            });
         });
-    });
-    *state = result
-}
+        *state = result
+    }
 
-#[inline(always)]
-fn add_constants(state: &mut [BaseElement; STATE_WIDTH], ark: &[BaseElement; STATE_WIDTH]) {
-    state.iter_mut().zip(ark).for_each(|(s, &k)| *s += k);
-}
+    #[inline(always)]
+    fn add_constants(state: &mut [BaseElement; STATE_WIDTH], ark: &[BaseElement; STATE_WIDTH]) {
+        state.iter_mut().zip(ark).for_each(|(s, &k)| *s += k);
+    }
 
-#[inline(always)]
-fn apply_sbox(state: &mut [BaseElement; STATE_WIDTH]) {
-    state.iter_mut().for_each(|v| {
-        let t2 = v.square();
-        let t4 = t2.square();
-        *v *= t2 * t4;
-    });
-}
+    #[inline(always)]
+    fn apply_sbox(state: &mut [BaseElement; STATE_WIDTH]) {
+        state.iter_mut().for_each(|v| {
+            let t2 = v.square();
+            let t4 = t2.square();
+            *v *= t2 * t4;
+        });
+    }
 
-#[inline(always)]
-fn apply_inv_sbox(state: &mut [BaseElement; STATE_WIDTH]) {
-    // compute base^10540996611094048183 using 72 multiplications per array element
-    // 10540996611094048183 = b1001001001001001001001001001000110110110110110110110110110110111
+    #[inline(always)]
+    fn apply_inv_sbox(state: &mut [BaseElement; STATE_WIDTH]) {
+        // compute base^10540996611094048183 using 72 multiplications per array element
+        // 10540996611094048183 = b1001001001001001001001001001000110110110110110110110110110110111
 
-    // compute base^10
-    let mut t1 = *state;
-    t1.iter_mut().for_each(|t| *t = t.square());
+        // compute base^10
+        let mut t1 = *state;
+        t1.iter_mut().for_each(|t| *t = t.square());
 
-    // compute base^100
-    let mut t2 = t1;
-    t2.iter_mut().for_each(|t| *t = t.square());
+        // compute base^100
+        let mut t2 = t1;
+        t2.iter_mut().for_each(|t| *t = t.square());
 
-    // compute base^100100
-    let t3 = exp_acc::<BaseElement, STATE_WIDTH, 3>(t2, t2);
+        // compute base^100100
+        let t3 = exp_acc::<BaseElement, STATE_WIDTH, 3>(t2, t2);
 
-    // compute base^100100100100
-    let t4 = exp_acc::<BaseElement, STATE_WIDTH, 6>(t3, t3);
+        // compute base^100100100100
+        let t4 = exp_acc::<BaseElement, STATE_WIDTH, 6>(t3, t3);
 
-    // compute base^100100100100100100100100
-    let t4 = exp_acc::<BaseElement, STATE_WIDTH, 12>(t4, t4);
+        // compute base^100100100100100100100100
+        let t4 = exp_acc::<BaseElement, STATE_WIDTH, 12>(t4, t4);
```

Comment:
This is pure refactor so it's okay, but it's also unsettling to read and perhaps to maintain: overwriting t4 makes the reviewer (and later coder?) wonder if there's a mistake.

After reading carefully, it looks like the first instance of t4 (100100100100) isn't directly used later, explaining why one might not need it. If there's no performance issue with using unique varnames, the reviewer wouldn't have to be suspicious.

Reply:
Good point! Fixed!